flybywiresim/a32nx
# feat(efb/ui): refueling UI override and initialization

Fixes #[issue_no]

## Summary of Changes

This PR completely overrides the use of MSFS's Fuel in-game menu, giving full and exclusive refuel/ defuel control to the EFB Dispatch fuel page.

Additionally, initial fuel quantity has been hardcoded and set randomly between 2,000 Kg & 3,000 Kg until persistence has been implemented.

IMPORTANT: Merging this PR will require our Comms&Marketing team to spread out the word about the change in functionality.

## Screenshots (if necessary)



## References





## Additional context



Discord username (if different from GitHub): TazX [Z+2]#405

## Testing instructions

1. Check Fuel in-game menu override:
    - Load a flight in any initial condition (parked, runway, in-flight)
    - Open the Fuel in-game menu and move around the fuel sliders
    - Although the sliders do move and seem to change fuel, in a matter of 1-3 seconds fuel levels should return to initial value
    - Go into the EFB->Dispatch->Fuel page
    - Change fuel level (refuel and defuel)
    - Check fuel levels do change accordingly
    
2. Check initial fuel quantity (random):
    - Restart the flight several times
    - Check fuel quantity is randomized between 2,000 Kg - 3,000 Kg (full outer tanks, zero center tank, rest within inner tanks)


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
